File: GW3/GWLogger/Frontend/DTOs/GatewayHistoricData.ts
class GatewayHistoricData
{
    CPU: LogStat[];
    Searches: LogStat[];
    PVs: LogStat[];
    Network: LogStat[];

    constructor(
        cpu: LogStat[],
        searches: LogStat[],
        pvs: LogStat[],
        network: LogStat[]
    ) {
        this.CPU = cpu;
        this.Searches = searches;
        this.PVs = pvs;
        this.Network = network;
    }

    public static CreateFromObject(o: any) {
        return new GatewayHistoricData(
            o.CPU ? o.CPU.map(v => LogStat.CreateFromObject(v)) : null,
            o.Searches ? o.Searches.map(v => LogStat.CreateFromObject(v)) : null,
            o.PVs ? o.PVs.map(v => LogStat.CreateFromObject(v)) : null,
            o.Network ? o.Network.map(v => LogStat.CreateFromObject(v)) : null
        );
    }

}

File: GW3/GWLogger/Live/GatewayHistoricData.cs
using System.Collections.Generic;

namespace GWLogger.Live
{
    public class GatewayHistoricData
    {

        public List<HistoricData> CPU { get; set; }
        public List<HistoricData> Searches { get; set; }
        public List<HistoricData> PVs { get; set; }
        public List<HistoricData> Network { get; set; }

    }
}